### PR TITLE
feat: Merge python3 branch to master

### PR DESCRIPTION
**Pull request by ki4070ma**
For https://github.com/appium/python-client/issues/468

This is step4 in below steps.

https://github.com/appium/python-client/issues/468#issuecomment-573276787
 - Step4. Merge pytnon3 branch to master

### CODE REVIEW

**Comment by KazuCocoa** [comment]
Is it safe to force push current `python3` branch as `master` rather than merge python3 branch into master?
I've already created a python2 branch from the latest master.

**Comment by mykola-mokhnach** [comment]
I'd rather squash and merge. Force pushes to master is not a good practice

**Comment by KazuCocoa** [comment]
Yeah, but fixing conflicts is tough work since the current master has been merged/sometimes rebased? to python3 branch. It needs to fix conflicts one by one (and make sure the fix has no wrong)
So, I thought so.

**Comment by ki4070ma** [comment]
I understand both points, but in this case it's not so tough task to resolve conflicts.
So I'm resolving conflicts.

File: appium/common/helper.py [hunk 1]
```diff
@@ -12,44 +12,25 @@
...
-    """
-
-    try:
-        return bytes(value, encoding)  # Python 3
-    except TypeError:
-        return value  # Python 2
-
-
-def extract_const_attributes(cls):
+def extract_const_attributes(cls: type) -> Dict:
```
Comment (mykola-mokhnach) [inline appium/common/helper.py:20]: Dict[str, Any]

File: appium/webdriver/common/touch_action.py [hunk 2]
```diff
@@ -126,12 +137,14 @@ def release(self):
...
 
-    def perform(self):
+    def perform(self: T) -> T:
         """Perform the action by sending the commands to the server to be operated upon
 
         Returns:
             `TouchAction`: self instance
         """
+        if self._driver is None:
+            raise TypeError('Set driver to constructor as a argument when to create the instance.')
```
Comment (mykola-mokhnach) [inline appium/webdriver/common/touch_action.py:147]: ValueError

File: appium/webdriver/extensions/android/activities.py [hunk 3]
```diff
@@ -12,15 +12,19 @@
...
 from selenium.webdriver.support.ui import WebDriverWait
 
 from appium.webdriver.mobilecommand import MobileCommand as Command
 
+T = TypeVar('T', bound='Activities')
+
 
 class Activities(webdriver.Remote):
-    def start_activity(self, app_package, app_activity, **opts):
+    def start_activity(self, app_package: str, app_activity: str, **opts: str) -> T:
```
Comment (mykola-mokhnach) [inline appium/webdriver/extensions/android/activities.py:27]: can `**opts` be of type `str`?

File: appium/webdriver/extensions/remote_fs.py [hunk 4]
```diff
@@ -13,50 +13,54 @@
...
 
 import base64
+from typing import Optional, TypeVar
 
 from selenium import webdriver
 from selenium.common.exceptions import InvalidArgumentException
 
 from ..mobilecommand import MobileCommand as Command
 
+T = TypeVar('T', bound='RemoteFS')
```
Comment (mykola-mokhnach) [inline appium/webdriver/extensions/remote_fs.py:23]: How about bounding all these basic types to AppiumWebDriver class? these mixins are anyway not used separately, but as a part of their "main" class

File: appium/webdriver/webdriver.py [hunk 5]
```diff
@@ -317,7 +319,7 @@ def find_elements(self, by=By.ID, value=None):
             'using': by,
             'value': value})['value'] or []
 
-    def create_web_element(self, element_id, w3c=False):
+    def create_web_element(self, element_id: int, w3c: bool = False) -> MobileWebElement:
```
Comment (mykola-mokhnach) [inline appium/webdriver/webdriver.py:322]: element id could be a string

File: docs/README.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+$bash generate.sh
+$cd python-client/docs/_build/html
+$python -m http.server 1234
+```
+
+Access to `http://localhost:1234` on web browser
+
+
+### How to deploy generated doc
+See https://github.com/ki4070ma/python-client-sphinx#how-to-deploy for now
```
Comment (mykola-mokhnach) [inline docs/README.md:21]: can we move that stuff into Appium org?

File: docs/README.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
+### How to generate doc
+
+```bash
+$cd python-client/docs
+$bash generate.sh
```
Comment (mykola-mokhnach) [inline docs/README.md:5]: should we mention make as a requirement?

File: docs/conf.py [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+#
+import os
+import sys
+
+sys.path.insert(0, os.path.abspath('../appium'))
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'Python client 1.0 beta'
```
Comment (mykola-mokhnach) [inline docs/conf.py:21]: why beta?

File: docs/make.bat [hunk 9]
```diff
@@ -0,0 +1,35 @@
+@ECHO OFF
```
Comment (mykola-mokhnach) [inline docs/make.bat:1]: wow, even for Windows

File: setup.py [hunk 10]
```diff
@@ -37,13 +37,10 @@
     packages=find_packages(include=['appium*']),
     license='Apache 2.0',
     classifiers=[
-        'Development Status :: 4 - Beta',
+        'Development Status :: 5 - Production/Stable',
```
Comment (mykola-mokhnach) [inline setup.py:40]: 👍

File: docs/README.md [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+### How to check generated doc
+
+```bash
+$cd python-client/docs
+$bash generate.sh
+$cd python-client/docs/_build/html
+$python -m http.server 1234
+```
+
+Access to `http://localhost:1234` on web browser
```
Comment (KazuCocoa) [inline docs/README.md:17]: What about making this link so that a user can open the page via the link?

File: docs/conf.py (hunk 8 above)
Comment (ki4070ma) [reply docs/conf.py:21]: Ah, I thought we will release 1.0beta at first, so just for it.
(It's not matched with https://github.com/appium/python-client/pull/526/files/6235b6a8d0339a92fa2fd2ff2e79d7c375e5143c#diff-2eeaed663bd0d25b7e608891384b7298R40 )

If any ideas, please let me know. > @mykola-mokhnach , @KazuCocoa 
I'll remove beta for this PR.

File: docs/README.md (hunk 6 above)
Comment (mykola-mokhnach) [reply docs/README.md:22]: yes, it makes sense to keep the stuff inside the org. @jlipps Could you please create the repository?

File: docs/conf.py (hunk 8 above)
Comment (mykola-mokhnach) [reply docs/conf.py:21]: I vote for removing beta. The stuff has been working for quite a while and is stable enough

File: docs/README.md (hunk 6 above)
Comment (jlipps) [reply docs/README.md:22]: repo now exists: https://github.com/appium/python-client-sphinx

Comment (mykola-mokhnach) [reply docs/README.md:22]: thanks Jonathan. 

@ki4070ma Would you mind to migrate the content there and update the links in the documentation?

Comment (ki4070ma) [reply docs/README.md:22]: I'll create another PR to update the URL here after uploading docs to below repo.

https://github.com/appium/python-client-sphinx

**Pull request merged**